core/drand_beacon_metrics.go

```diff
@@ -1,28 +1,32 @@
 package core
 
 import (
-	"context"
 	"errors"
+	"fmt"
 	"net/http"
 
+	"github.com/drand/drand/log"
 	"github.com/drand/drand/net"
 )
 
-// PeerMetrics exports a handler for retreiving metric information from group peers
-func (bp *BeaconProcess) PeerMetrics(c context.Context) (map[string]http.Handler, error) {
+// GroupMetrics exports a handler for retrieving metric information from group peers
+// It's a function so that it can be called lazily when nodes join/leave (e.g. during
+// reshares)
+func (bp *BeaconProcess) GroupMetrics() (map[string]http.Handler, error) {
 	if bp.group == nil {
-		return nil, errors.New("no group yet")
+		return nil, errors.New(fmt.Sprintf("no group yet for beacon %s", bp.beaconID))
 	}
 
 	pc := bp.privGateway.ProtocolClient
 	hc, ok := pc.(net.HTTPClient)
 	if !ok {
-		return nil, errors.New("implementation does not support metrics")
+		return nil, errors.New("ProtocolClient implementation does not support metrics")
```

Comment:
You'll need a `// nolint:golint` of start with a small letter to pass the linter

Reply:
👍🏻